java/vector/src/main/java/org/apache/arrow/vector/FieldVector.java

```diff
@@ -52,6 +52,13 @@
   void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers);
 
   /**
+   * load data into vector using just the Buffers flatbuffer and backing ArrowBuf
+   * @param buffersIterator
+   * @param buf
+   */
+  void loadFieldBuffers(BuffersIterator buffersIterator, ArrowBuf buf);
```

Comment:
Why do we need to introduce this method in the API.
How is it different from just slicing buf appropriately based on BuffersIterator and call the existing loadFieldBuffers?